cmd/imgcollector/image/artifact.go

```diff
@@ -128,6 +128,11 @@ func (a Artifact) Inspect(ctx context.Context) (*ArtifactReference, error) {
 		return nil, fmt.Errorf("analyze error: %w", err)
 	}
 
+	if cachedOSInfo != nil {
+		// we use cached artifactInfo, because a.inpsect did not craete artifact info.
+		artifactInfo = cachedOSInfo
```

Comment:
Maybe rename this variable and method to getCachedArtifactInfo ? Now it's confusing since we have os info and artifact info